Best practices/Serialization & Deserialization.md

```diff
@@ -0,0 +1,106 @@
+   
+    var options = new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull }; 
+        
+**AllowTrailingCommas**: When set to `true`, the deserializer will tolerate trailing commas in the JSON.
+    
+    var options = new JsonSerializerOptions { AllowTrailingCommas = true }; 
+    
+**MaxDepth**: Sets the maximum depth allowed when reading or writing JSON, protecting against deeply nested structures.
+    
+    var options = new JsonSerializerOptions { MaxDepth = 10 }; 
+    
+**ReferenceHandler**: Determines how to handle references, which can be useful for dealing with circular references or preserving object references.
+    
+    var options = new JsonSerializerOptions { ReferenceHandler = ReferenceHandler.Preserve };
+    
+**Converters**: You can add custom converters to handle specific serialization or deserialization scenarios that aren't covered by the default behavior.
+    
+    options.Converters.Add(new MyCustomConverter()); 
+    
+These are just some of the many options available in `JsonSerializerOptions`. By understanding and utilizing these options, developers can fine-tune the behavior of `System.Text.Json` to match the specific needs of their applications.
+
+### Setting up JsonSerializerOptions globally
+
+The `AddJsonOptions` extension method allows you to configure serialization settings globally for all controllers in an ASP.NET Core application.  You typically do this in the `Program.cs` while configuring MVC services. Example:
+
+	services.AddControllers()
+		.AddJsonOptions(options => { ... });
+
+## Source Generators
+
+Starting from .NET 6, `System.Text.Json` introduced source generators, enhancing serialization performance. This feature can operate in two primary modes: gathering metadata and optimizing serialization. By default it utilises both modes.
+
+### Basic Setup:
+
+To use source generation with default settings:
+-   Create a `partial` class that inherits from `JsonSerializerContext`.
+-   Mark the types you wish to serialize or deserialize with `JsonSerializableAttribute`.
+-   Invoke a `JsonSerializer` method that:
+    -   Accepts a `JsonTypeInfo<T>` instance, or
+    -   Accepts a `JsonSerializerContext` instance, or
+	 -  Uses a a `JsonSerializerOptions` instance and you've set its `JsonSerializerOptions.TypeInfoResolver` property to the `Default` property of the context type
+
+
+### Examples:
+
+Consider a `WeatherForecast` class:
+
+    public class WeatherForecast 
+    {
+        public DateTime Date { get; set; }
+        public int TemperatureCelsius { get; set; }
+        public string? Summary { get; set; }
+    } 
+
+To enable source generation for this class:
+
+    [JsonSerializable(typeof(WeatherForecast))]
+    public partial class MyJsonContext : JsonSerializerContext { } 
```

Comment:
If this is a context only for the `WeatherForecast` class, we should name it accordingly.

Reply:
I see, but since this was not apparent to me as a reader, then maybe we should add this info here.